EosioSwiftEcc/Recover/EccRecoverKey.swift

```diff
@@ -43,7 +43,9 @@ public class EccRecoverKey {
         EC_KEY_set_group(key, group)
 
         var recoveredPubKeyHex = ""
-        privateKey.withUnsafeBytes { (pkbytes: UnsafePointer<UInt8>) -> Void in
+        privateKey.withUnsafeBytes { rawBufferPointer in
+            let bufferPointer = rawBufferPointer.bindMemory(to: UInt8.self)
+            guard let pkbytes = bufferPointer.baseAddress else { return }
```

Comment:
Since these functions throw we could throw if we can't get the `baseAddress` of the data.  The only reason I can think of that it would be nil would be if Data is empty.  I'll defer to the other developers who have worked on this project if there is an appropriate error that could be thrown here.  The way it's written now should be fine since you can't recover a public key from an empty Data anyways but just checking if we want to throw here.

Reply:
yeah, I think it would be good to provide an opportunity for error handling

Reply:
The SDK currently throws EosioError with codes of `.keySigningError` already in this class so I'd say its appropriate to add a throw here with that code and a reason of something like can't get the base address of the data.

Reply:
@opi-smccoole do you think it makes sense to add this as part of an enhancement ticket? or should I go ahead and squeeze it in here.